### PR TITLE
Update the `iris.experimental` module

### LINKED ISSUE #3515: Tidy iris.experimental module
Identify and pull through any experimental functionality from `iris.experimental` into the core of `iris`, and also retire any obsolete functionality.

---
### Acceptance Criteria
- [x] Remove `iris.experimental.concatenate` (see #3523)
- [x] Rationalise `iris.experimental.equalise_cubes` (see #3527)
  - [x] Move `iris.experimental.equalise_cubes.equalise_attributes` to `iris.util`
  - [x] Replace `iris.experimental.equalise_cubes.equalise_attributes` with an exception, e.g., see [here](https://github.com/SciTools/iris/blob/v2.2.1/lib/iris/experimental/concatenate.py#L43-L48)
- [x] Review outstanding `iris.experimental` functionality
 - NB: Must include identify all that are used operationally, (def includes areaweighted regridding). as they must not be deprecated.

### PR DESCRIPTION
**Pull request by trexfeathers**
### Description

Standard 'spring cleaning' to make sure the contents of `iris.experimental` belong there (see https://github.com/SciTools/iris/issues/3515#issuecomment-1309054877).

Requires SciTools/iris-test-data#83.

Closes #3515.

(For moving something out of `iris.experimental`, I copied the style in #3390).

---
[Consult Iris pull request check list]( https://scitools-iris.readthedocs.io/en/latest/developers_guide/contributing_pull_request_checklist.html)

### CODE REVIEW

File: lib/iris/experimental/animate.py [hunk 1]
```diff
@@ -6,118 +6,32 @@
...
-
-    return animation.FuncAnimation(
-        fig, update, frames=frames, fargs=(cubes, vmin, vmax, coords), **kwargs
+    msg = (
+        "The function 'iris.experimental.animate.animate()' has been moved, "
+        "and is now at 'iris.plot.animate()'.\n"
+        "Please replace 'iris.experimental.animate.animate' with "
+        "'iris.plot.animate'."
     )
+    raise Exception(msg)
```
Comment (bjlittle) [inline lib/iris/experimental/animate.py:37]: @trexfeathers Perhaps raise a `DeprecationWarning` instead?

**Review by bjlittle — CHANGES_REQUESTED** [review]
@trexfeathers Thanks, a lovely piece of housekeeping :partying_face: 

Just a minor comment to address regarding the choice of exception to raise.

Also, the `benchmark` GHA makes reference to `IRIS_TEST_DATA_VERSION: "2.15"`... did you want to update that also?

File: lib/iris/experimental/animate.py (hunk 1 above)
Comment (trexfeathers) [reply lib/iris/experimental/animate.py:37]: I'm all for being more specific, but it seems important to actually exit with an error here to be as clear as possible. What would you think of the below?

```suggestion
    with warnings.simplefilter("error", category=IrisDeprecation):
        iris._deprecation.warn_deprecated(msg)
```

Comment (bjlittle) [reply lib/iris/experimental/animate.py:37]: TBH it's fine... previous moves of functionality out of experimental have simply raised an `Exception`, so let's just roll with that :+1:

Comment (trexfeathers) [reply lib/iris/experimental/animate.py:37]: Or have the warning and an `Exception` straight after

**Review by bjlittle — APPROVED** [review]

**Comment by trexfeathers** [comment]
All kinds of done! Thanks again 😊

**Pull request merged**